src/transform.rs

```
@@ -1,46 +1,66 @@
+        regex,
+    },
+    anyhow::Result,
+};
 
-use crate::errors;
-
-/// This function takes the rust code as input
-/// and returns the code with DejaGnu directive
+/// Transform code to `DejaGnu` format
 pub fn transform_code(code: &str, stderr_file: Option<&str>) -> Result<String> {
+    // Load the rustc error messages, codes, lines and relative line numbers
     let errors = errors::load_error(code, stderr_file);
+    // For storing the transformed code
     let mut new_code = String::new();
+    let additional_options = parse_additional_options(code);
 
     let mut line_num = 1;
+    // finding the respective line number and adding the error code
     for line in code.lines() {
         let mut new_line = line.to_string();
-        // TODO: This is not the efficient way to find respective line number
-        for error in errors.iter() {
-            if (error.line_num as i32 - error.relative_line_num) != line_num {
-                continue;
+
+        if is_header_line(line) {
+            for header in additional_options.iter() {
+                if header.line_number != line_num {
+                    continue;
+                }
+                new_line = header.dejagnu_header.to_string();
+                break;
             }
-            // In rustc test suites, the error directive is
-            // on the same line or the next line not on the previous line
-            // For the error on the next line
-            if error.relative_line_num != 0 {
-                new_line = format!("{}", error);
-            } else {
-                // For the error on the same line
-                let re: OnceCell<Regex> = OnceCell::new();
+        } else {
+            // TODO: This is not the efficient way to find respective line number
+            for error in errors.iter() {
+                // Checking the original line number
+                if (error.line_num as i32 - error.relative_line_num) != line_num as i32 {
```

I'd prefer if you could use either the `From` trait or the `TryFrom` trait (or their `Into` counterpart) instead of a cast because I don't like silent errors that morph into bugs.

---

Sure, this surely helps to catch undesired behaviors :)